wagtail/contrib/modeladmin/views.py

```diff
@@ -136,19 +144,15 @@ def media(self):
             js=self.model_admin.get_form_view_extra_js()
         )
 
-    def get_context_data(self, **kwargs):
-        instance = self.get_instance()
-        edit_handler = self.get_edit_handler()
-        form = self.get_form()
-        edit_handler = edit_handler.bind_to(
-            instance=instance, request=self.request, form=form)
-        context = {
+    def get_context_data(self, form=None, **kwargs):
+        if form is None:
+            form = self.get_form()
+        kwargs.update({
```

Comment:
I think we need to keep the previous pattern here of defining a dictionary, then updating it with `kwargs` - otherwise, subclasses lose the ability to override these values by providing alternative values using `kwargs`.

Reply:
I tried to keep that in mind actually and I think it is still possible at some point:
* `form`: an alternative value can still be defined in keyword arguments of `get_context_data` - and it will now only be created if its value is None.
* `is_multipart`: as it is directly linked to `form`, I am not sure if it is safe to override this value here.
* `edit_handler`: it is indeed not possible anymore to override its value in `get_context_data`. However, one can now instead provide an alternative value either in `get_edit_handler` or in `edit_handler` object's property. I think providing an alternative value only in `get_context_data` should be discouraged since this new possibilities ensure at least consistency. What do you think?

Reply:
@jeromelebleu This is more about backwards compatibility than anything - Currently, if a user was doing something like the following, it would no longer work as expected:

```
MyEditView(EditView):
    def get_context_data(self, **kwargs):
        return super().get_context_data(
            edit_handler=some_alternative_handler
        )
```

There are a lot of things that vary throughout wagtail, but this pattern is something we have tried to stick to elsewhere when it comes to `get_context_data()` - it would be good to keep this element as it was for consistency.

Reply:
Thanks for this explanations, I understand your point and totally agree in fact with the backward compatibility argument. I have made requested changes!